magento/magento2
# 2.4 develop bugfix Url rewrite for new category doesn't generate correctly

### Description (*)
The category URL rewrite is generated for all store view

### Related Pull Requests
 if we set store id as 0 it will create category URL rewrite is generated for all store view

### Fixed Issues (if relevant)

1. Fixes magento/magento2#36936

### Manual testing scenarios (*)
Create a new store view.

Go to Stores >> Configuration >> General >> Search Engine Optimization >> Set "Yes" in Use Web Server Rewrites.

Go to System >> Data transfer >>Import ->Import a file of products and add new category in the Category column of the file (make sure that category should not be created in the backend).
[catalog_product.csv](https://github.com/magento/magento2/files/10882084/catalog_product.csv)

Now, verify the url rewrite generated for the storeview of the category in Marketing >> Seo & Search >> Url Rewrite.

Expected result: The category URL rewrite is generated for all store view.
Actual result: The category URL rewrite is generated for only the default store view of the store group.


### Contribution checklist (*)
 - [*] Pull request has a meaningful description of its purpose
 - [*] All commits are accompanied by meaningful commit messages
 - [*] All new or changed code is covered with unit/integration tests (if applicable)
 - [*] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [*] All automated tests passed successfully (all builds are green)
